sdk/test-utils/recorder-new/src/utils/delay.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { isPlaybackMode } from "./utils";
+
+/**
+ * Usage - `await delay(<milliseconds>)`
+ * This `delay` has no effect if the `TEST_MODE` is `"playback"`.
+ * If the `TEST_MODE` is not `"playback"`, `delay` is a wrapper for setTimeout that resolves a promise after t milliseconds.
+ *
+ * @param {number} milliseconds The number of milliseconds to be delayed.
+ * @returns {Promise<T>} Resolved promise
+ */
+export function delay(milliseconds: number): Promise<void> | null {
+  return isPlaybackMode() ? null : new Promise((resolve) => setTimeout(resolve, milliseconds));
+}
```

Comment:
Can return `Promise<void> | void`
```suggestion
export function delay(milliseconds: number): Promise<void> | void {
  if (isPlaybackMode()) {
    return;
  }
  return new Promise((resolve) => setTimeout(resolve, milliseconds));
}
```